topostats/grains.py

```diff
@@ -742,3 +795,741 @@ def keep_largest_labelled_region(
         sizes = np.array([(labelled_image == label).sum() for label in range(1, labelled_image.max() + 1)])
         # Keep only the largest region
         return np.where(labelled_image == sizes.argmax() + 1, labelled_image, 0).astype(bool)
+
+    @staticmethod
+    def flatten_multi_class_tensor(grain_mask_tensor: npt.NDArray) -> npt.NDArray:
+        """
+        Flatten a multi-class image tensor to a single binary mask.
+
+        Parameters
+        ----------
+        grain_mask_tensor : npt.NDArray
+            Multi class grain mask tensor tensor of shape (N, N, C).
+
+        Returns
+        -------
+        npt.NDArray
+            Combined binary mask of all but the background class (:, :, 0).
+        """
+        assert len(grain_mask_tensor.shape) == 3, f"Tensor not 3D: {grain_mask_tensor.shape}"
+        # Convert to binary in case there are multiple hits in the same pixel. We dont want to have 2s, 3s etc
+        # because this would cause issues in labelling and cause grains in grains.
+        return np.sum(grain_mask_tensor[:, :, 1:], axis=-1).astype(bool)
+
+    @staticmethod
+    def get_multi_class_grain_bounding_boxes(grain_mask_tensor: npt.NDArray) -> dict:
+        """
+        Get the bounding boxes for each grain in a multi-class image tensor.
+
+        Finds the bounding boxes for each grain in a multi-class image tensor. Grains can span multiple classes, so the
+        bounding boxes are found for the combined binary mask of contiguous grains across all classes.
+
+        Parameters
+        ----------
+        grain_mask_tensor : npt.NDArray
+            3-D Numpy array of grain mask tensor.
+
+        Returns
+        -------
+        dict
+            Dictionary of bounding boxes indexed by grain number.
+        """
+        # Get the flattened mask
```

Comment:
I must be getting old (or slightly better at reading code) as I feel these comments are extraneous given we have expressive variable and method names which make it clear what is happening at each step.